src/components/DonutProgress/DonutProgress.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+
+import { DEFAULT_SIZE, DEFAULT_STROKE_WIDTH } from './DonutProgress.constants';
+import { useDonutColor } from './DonutProgress.hooks';
+import { DonutContainer, DonutText } from './DonutProgress.styles';
+
+import type { DonutProgressProps } from './DonutProgress.types';
+
+export const DonutProgress: React.FC<DonutProgressProps> = ({
+  percentage,
+  color = 'auto',
+  size = DEFAULT_SIZE,
+  strokeWidth = DEFAULT_STROKE_WIDTH,
+  animated = true,
+}) => {
+  const radius = (size - strokeWidth) / 2;
+  const circumference = 2 * Math.PI * radius;
+  const offset = circumference - (percentage / 100) * circumference;
+
+  const computedColor = useDonutColor(color, percentage);
+
+  return (
+    <DonutContainer size={size}>
+      <svg width={size} height={size}>
+        <circle
+          stroke="#6b6868ff"
+          fill="transparent"
+          strokeWidth={strokeWidth}
+          r={radius}
+          cx={size / 2}
+          cy={size / 2}
+        />
+        <circle
+          stroke={computedColor}
+          fill="transparent"
+          strokeWidth={strokeWidth}
+          strokeDasharray={`${circumference} ${circumference}`}
```

Comment:
strokeDasharray is provided as a template string including both values; that's acceptable. However transitions/animations should target only properties that will change (stroke-dashoffset). Animating stroke color can be expensive; if color changes are expected keep it, otherwise remove or shorten color animation.

**Sugerencia:**
```typescript
style={{ transition: animated ? 'stroke-dashoffset 0.8s ease' : 'none', ... }}
```